src/libs/pulumi-cli.ts

```diff
@@ -8,3 +14,46 @@ export async function isAvailable(): Promise<boolean> {
 export async function run(...args: string[]): Promise<void> {
   await exec.exec(`pulumi`, args, true);
 }
+
+export async function downloadCli(range: string): Promise<void> {
```

Comment:
what happens if there is already a version installed - will this new install overwrite that?

Reply:
AFAIK no, both will be installed, both in `PATH`. We could validate if it is available and exit, or potentially deterministically delete (expect it to be in `.pulumi`?)

Reply:
We could also fail if `pulumi` is installed?

Reply:
So for this, we will ensure we use the version that the action installs internally